packages/botonic-react/src/components/multichannel/whatsapp/markdown.ts

```diff
@@ -1,105 +1,66 @@
+
+// Convert markdown to WhatsApp and Facebook format
+export function convertToMarkdownMeta(text: string): string {
   const textWithItalicAndBold = replaceItalicAndBold(text)
+
   return replaceMarkdownLinks(textWithItalicAndBold)
 }
 
-export function replaceItalicAndBold(text: string) {
-  const textNormalized = normalizeMarkdown(text)
-  const matches = textNormalized.match(MARKDOWN_BOLD_OR_ITALIC_REGEX)
-  if (matches) {
-    const matchesResult = matches.map(match => {
-      if (match.startsWith(MARKDOWN_BOLD_OPTION_1)) {
-        return replaceAllOccurrences(
-          match,
-          MARKDOWN_BOLD_OPTION_1,
-          MARKDOWN_WHATSAPP_BOLD
-        )
-      }
-
-      if (match.startsWith(MARKDOWN_BOLD_OPTION_2)) {
-        return replaceAllOccurrences(
-          match,
-          MARKDOWN_BOLD_OPTION_2,
-          MARKDOWN_WHATSAPP_BOLD
-        )
-      }
-
-      if (match.startsWith(MARKDOWN_ITALIC_OPTION_1)) {
-        return replaceAllOccurrences(
-          match,
-          MARKDOWN_ITALIC_OPTION_1,
-          MARKDOWN_WHATSAPP_ITALIC
-        )
-      }
-
-      return match
-    })
-
-    let textWhatsapp = textNormalized
-    for (let i = 0; i < matches.length; i++) {
-      textWhatsapp = replaceAllOccurrences(
-        textWhatsapp,
-        matches[i],
-        matchesResult[i]
-      )
-    }
+function replaceItalicAndBold(text: string) {
+  const normalizedText = normalizeMarkdown(text)
+  const boldAndItalicText = convertNormalizedToWhatsApp(normalizedText)
 
-    return textWhatsapp
-  }
-
-  return text
+  return boldAndItalicText
 }
 
-function normalizeMarkdown(text: string) {
-  text = replaceBoldItalic(text, MARKDOWN_BOLD_AND_ITALIC_OPTION1)
-  text = replaceBoldItalic(text, MARKDOWN_BOLD_AND_ITALIC_OPTION2)
-  text = replaceBoldItalic(text, MARKDOWN_BOLD_AND_ITALIC_OPTION3)
+function normalizeMarkdown(text: string): string {
+  // Normalize bold
+  text = text.replace(
+    REGEX_MARKDOWN_BOLD,
+    `${NORMALIZED_BOLD}$2${NORMALIZED_BOLD}`
+  )
+  // Normalize italic
+  text = text.replace(
+    REGEX_MARKDOWN_ITALIC,
+    `${NORMALIZED_ITALIC}$2${NORMALIZED_ITALIC}`
+  )
+
   return text
 }
 
-function replaceBoldItalic(text: string, regex: RegExp) {
-  return text.replace(
-    regex,
-    (
-      match: string,
-      markdownOpen: string,
-      textInsideMarkdown: string,
-      markdownClose: string
-    ) => {
-      if (match.startsWith(markdownOpen) && match.endsWith(markdownClose)) {
-        return `${MARKDOWN_NORMALIZED_BOLD_ITALIC_OPEN}${textInsideMarkdown}${MARKDOWN_NORMALIZED_BOLD_ITALIC_CLOSE}`
-      }
-      return match
-    }
+function convertNormalizedToWhatsApp(text: string): string {
+  // convert &%BOLD%&text&%BOLD%& to *text*
+  text = text.replace(
+    REGEX_NORMALIZED_BOLD,
+    `${WHATSAPP_BOLD}$1${WHATSAPP_BOLD}`
+  )
+  // convert &%ITALIC%&text&%ITALIC%& to _text_
+  text = text.replace(
+    REGEX_NORMALIZED_ITALIC,
+    `${WHATSAPP_ITALIC}$1${WHATSAPP_ITALIC}`
   )
```

Comment:
while I understand this is working correctly, I would do some changes to make it more readable for us in the future, just a possible one suggested by IA:
```
// Constants for Markdown and Normalized/WhatsApp symbols
const REGEX_MARKDOWN_BOLD = /\*\*(.+?)\*\*/g;
const REGEX_MARKDOWN_ITALIC = /_(.+?)_/g;
const NORMALIZED_BOLD = "&%BOLD%&";
const NORMALIZED_ITALIC = "&%ITALIC%&";

const REGEX_NORMALIZED_BOLD = /&%BOLD%&(.+?)&%BOLD%&/g;
const REGEX_NORMALIZED_ITALIC = /&%ITALIC%&(.+?)&%ITALIC%&/g;
const WHATSAPP_BOLD = "*";
const WHATSAPP_ITALIC = "_";

// Helper function for formatting conversion
function applyFormatting(
  text: string,
  regex: RegExp,
  openingSymbol: string,
  closingSymbol: string = openingSymbol
): string {
  return text.replace(regex, `${openingSymbol}$1${closingSymbol}`);
}

// Normalize Markdown to custom format
function normalizeMarkdown(text: string): string {
  text = applyFormatting(text, REGEX_MARKDOWN_BOLD, NORMALIZED_BOLD);
  text = applyFormatting(text, REGEX_MARKDOWN_ITALIC, NORMALIZED_ITALIC);
  return text;
}

// Convert custom format to WhatsApp format
function convertNormalizedToWhatsApp(text: string): string {
  text = applyFormatting(text, REGEX_NORMALIZED_BOLD, WHATSAPP_BOLD);
  text = applyFormatting(text, REGEX_NORMALIZED_ITALIC, WHATSAPP_ITALIC);
  return text;
}
```

Reply:
It cannot be as you propose because in the applyFormatting function it would need another input parameter for the $1 that goes between the openingSymbol and the closingSymbol.